aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsFluentClientDecorator.kt

```diff
@@ -110,7 +111,8 @@ private class AwsFluentClientExtensions(private val types: Types) {
 
                 ##[cfg(any(feature = "rustls", feature = "native-tls"))]
                 pub fn from_conf(conf: crate::Config) -> Self {
-                    let client = #{aws_hyper}::Client::https();
+                    let retry_config = conf.retry_config.as_ref().cloned().unwrap_or_default();
+                    let client = #{aws_hyper}::Client::https().with_retry_config(retry_config.into());
```

Comment:
might be worth going through `from_conf_conn` here so that there's only one place we need to remember to use the retry_config